File: 0485-max-consecutive-ones/0485-max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int max=0, lastMax=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==1){
                max++;
            }else{
                max=0;
            }
            if(lastMax<max){
                lastMax=max;
            }
        }
        return lastMax;   
    }
}

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_consecutive_ones=0
        curr_consecutive_ones=0
        for i in range(len(nums)):
            if nums[i]==1:
                curr_consecutive_ones+=1
            else:
                max_consecutive_ones=max(max_consecutive_ones, curr_consecutive_ones)
                curr_consecutive_ones=0
        return max(max_consecutive_ones, curr_consecutive_ones)